workitem/workitem_repository.go

```diff
@@ -631,42 +650,56 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, spaceID uuid.UUID, up
 	return ConvertWorkItemStorageToModel(wiType, wiStorage)
 }
 
-// Create creates a new work item in the repository
-// returns BadParameterError, ConversionError or InternalError
-func (r *GormWorkItemRepository) Create(ctx context.Context, spaceID uuid.UUID, typeID uuid.UUID, fields map[string]interface{}, creatorID uuid.UUID) (*WorkItem, error) {
-	defer goa.MeasureSince([]string{"goa", "db", "workitem", "create"}, time.Now())
-
-	wiType, err := r.witr.Load(ctx, typeID)
-	if err != nil {
-		return nil, errors.NewBadParameterError("typeID", typeID)
-	}
-
+// CheckWITypeBelongsToSpace returns true if the given workitem type (wit) belongs to the same space template as the space (spaceID)
+func (r *GormWorkItemRepository) CheckWITypeBelongsToSpace(ctx context.Context, wit *WorkItemType, spaceID uuid.UUID) (bool, error) {
```

Comment:
[MINOR] Function name should be corrected. Work Item Type does not "belong to space"

Reply:
@DhritiShikhar I've addressed that in 038c56e4 and renamed the function to `CheckTypeAndSpaceShareTemplate`.